bin/propolis-server/src/lib/server.rs

```diff
@@ -953,6 +955,77 @@ async fn instance_issue_crucible_snapshot_request(
     Ok(HttpResponseOk(()))
 }
 
+/// Issues a volume_construction_request replace to a crucible backend.
+#[endpoint {
+    method = PUT,
+    path = "/instance/disk/{id}/vcr",
+}]
+async fn instance_issue_crucible_vcr_request(
+    rqctx: RequestContext<Arc<DropshotEndpointContext>>,
+    path_params: Path<api::VCRRequestPathParams>,
+    request: TypedBody<api::InstanceVCRReplace>,
+) -> Result<HttpResponseOk<()>, HttpError> {
+    let path_params = path_params.into_inner();
+    let request = request.into_inner();
+    let new_vcr_json = request.vcr_json;
+    let disk_name = request.name;
+    let log = rqctx.log.clone();
+
+    // Get the instance spec for storage backend from the disk name.  We use
+    // the VCR stored there to send to crucible along with the new VCR we want
+    // to replace it.
+    let vm_controller = rqctx.context().vm().await?;
+    let mut spec = vm_controller.instance_spec().await;
+    let VersionedInstanceSpec::V0(v0_spec) = &mut *spec;
+
+    let (readonly, old_vcr_json) = {
+        let bes = &v0_spec.backends.storage_backends.get(&disk_name);
+        if let Some(StorageBackendV0::Crucible(bes)) = bes {
+            (bes.readonly, &bes.request_json)
+        } else {
+            let s = format!("Crucible backend for {:?} not found", disk_name);
+            return Err(HttpError::for_not_found(Some(s.clone()), s));
+        }
+    };
+
+    // Get the crucible backend so we can call the replacement method on it.
+    let crucible_backends = vm_controller.crucible_backends();
+    let backend = crucible_backends.get(&path_params.id).ok_or_else(|| {
+        let s = format!("No crucible backend for id {}", path_params.id);
+        HttpError::for_not_found(Some(s.clone()), s)
+    })?;
+
+    slog::info!(
+        log,
+        "{:?} {:?} replace {:?} with {:?}",
+        disk_name,
+        path_params.id,
+        old_vcr_json,
+        new_vcr_json,
+    );
```

Comment:
Optional: I've begun evangelizing for putting values into log messages using KV pairs instead of interleaving them into messages:

```suggestion
    slog::info!(
        log,
        "replacing downstairs for Crucible disk";
        "disk_name" => %disk_name,
        "volume_id" => %path_params.id,
        "old_vcr" => %old_vcr_json,
        "new_vcr" => %new_vcr_json,
    );
```

When examining logs you can then filter on the specific fields you're interested in instead of having to search for them in the message body; e.g. you can do `looker -c 'r.volume_id?.contains("<guid>")' and find any record that has a structured volume ID set in it irrespective of how its message was formatted.

Then again, `looker -c 'r.msg.contains("<guid>")` is not *that* much more onerous, so this is your call.

Reply:
I took the log line out, but I'll remember that trick for future logging.